### ISSUE TITLE
[Bug]: When the string data type in int field, the bulk insert can not directly failed, but keep in Started status for a long time

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#21818]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20230119-60e3e661
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):Pulsar    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
self = <test_bulk_insert.TestBulkInsertInvalidParams object at 0x7fe38099cca0>
is_row_based = True, dim = 8, entities = 10

    @pytest.mark.tags(CaseLabel.L3)
    @pytest.mark.parametrize("is_row_based", [True])
    @pytest.mark.parametrize("dim", [8])
    @pytest.mark.parametrize("entities", [10])
    def test_data_type_string_on_int_pk(self, is_row_based, dim, entities):
        """
        collection schema: default multi scalars
        data file: json file with one of entities has string on int pk
        Steps:
        1. create collection
        2. import data with is_row_based=False
        3. verify import failed
        """
        files = prepare_bulk_insert_json_files(
            minio_endpoint=self.minio_endpoint,
            bucket_name=self.bucket_name,
            is_row_based=is_row_based,
            rows=entities,
            dim=dim,
            auto_id=False,
            data_fields=default_multi_fields,
            err_type=DataErrorType.str_on_int_pk,
            force=True,
        )
    
        self._connect()
        c_name = cf.gen_unique_str("bulk_insert")
        # TODO: add string pk
        fields = [
            cf.gen_int64_field(name=df.pk_field, is_primary=True),
            cf.gen_float_vec_field(name=df.vec_field, dim=dim),
            cf.gen_int32_field(name=df.int_field),
            cf.gen_string_field(name=df.string_field),
            cf.gen_bool_field(name=df.bool_field),
            cf.gen_float_field(name=df.float_field),
        ]
        schema = cf.gen_collection_schema(fields=fields, auto_id=False)
        self.collection_wrap.init_collection(c_name, schema=schema)
        # import data
        task_id, _ = self.utility_wrap.do_bulk_insert(
            collection_name=c_name, files=files
        )
        logging.info(f"bulk insert task ids:{task_id}")
        success, states = self.utility_wrap.wait_for_bulk_insert_tasks_completed(
            task_ids=[task_id], timeout=90
        )
        log.info(f"bulk insert state:{success}")
        assert not success
        failed_reason = f"illegal numeric value"
        for state in states.values():
>           assert state.state_name in ["Failed", "Failed and cleaned"]
E           AssertionError: assert 'Started' in ['Failed', 'Failed and cleaned']
E            +  where 'Started' = <Bulk insert state:\n    - taskID          : 438857678255370314,\n    - state           : Started,\n    - row_count       : 0,\n    - infos           : {'files': 'str_on_int_pk_row_cust_float_vectors_multi_scalars_8d_10_0.json', 'collection': 'bulk_insert_w4TcAnFa', 'partition': '_default', 'failed_reason': ''},\n    - id_ranges       : [],\n    - create_ts       : 2023-01-19 14:57:48\n>.state_name
```

### Expected Behavior

expected the status is Failed or Failed and cleaned

### Milvus Log

[bulk-insert-test-01-19-15-03.zip](https://github.com/milvus-io/milvus/files/10454276/bulk-insert-test-01-19-15-03.zip)

**Comment by zhuwenxing** [comment]
It would cause datanode panic
```
[2023/02/07 07:56:43.991 +00:00] [INFO] [importutil/import_wrapper.go:256] ["import wrapper: begin import"] [filePaths="[str_on_int_pk_row_cust_float_vectors_multi_scalars_8d_10_0.json]"] [options="{\"OnlyValidate\":false,\"TsStartPoint\":0,\"TsEndPoint\":18446744073709551615,\"IsBackup\":false}"]
[2023/02/07 07:56:43.995 +00:00] [INFO] [importutil/import_wrapper.go:278] ["import wrapper:  row-based file "] [filePath=str_on_int_pk_row_cust_float_vectors_multi_scalars_8d_10_0.json] [fileType=.json]
[2023/02/07 07:56:43.998 +00:00] [INFO] [importutil/json_parser.go:106] ["JSON parser: reset bufRowCount"] [sizePerRecord=305] [bufRowCount=1024]
[2023/02/07 07:56:44.001 +00:00] [INFO] [datanode/data_node.go:973] ["DataNode finish import request"] ["task ID"=439288669910475186]
panic: interface conversion: interface {} is string, not json.Number

goroutine 247 [running]:
[github.com/milvus-io/milvus/internal/util/importutil.(*JSONRowConsumer).Handle](http://github.com/milvus-io/milvus/internal/util/importutil.(*JSONRowConsumer).Handle)(0xc0006d40e0, {0xc001b98000, 0xa, 0x92?})
        /go/src/[github.com/milvus-io/milvus/internal/util/importutil/json_handler.go:251](http://github.com/milvus-io/milvus/internal/util/importutil/json_handler.go:251) +0x1cac
[github.com/milvus-io/milvus/internal/util/importutil.(*JSONParser).ParseRows](http://github.com/milvus-io/milvus/internal/util/importutil.(*JSONParser).ParseRows)(0xc001b0a240, 0xc001b96c60, {0x41b5820, 0xc0006d40e0})
        /go/src/[github.com/milvus-io/milvus/internal/util/importutil/json_parser.go:238](http://github.com/milvus-io/milvus/internal/util/importutil/json_parser.go:238) +0xfcb
[github.com/milvus-io/milvus/internal/util/importutil.(*ImportWrapper).parseRowBasedJSON](http://github.com/milvus-io/milvus/internal/util/importutil.(*ImportWrapper).parseRowBasedJSON)(0xc0018b9d40, {0xc0015ce300, 0x3f}, 0x0)
        /go/src/[github.com/milvus-io/milvus/internal/util/importutil/import_wrapper.go:455](http://github.com/milvus-io/milvus/internal/util/importutil/import_wrapper.go:455) +0x43d
[github.com/milvus-io/milvus/internal/util/importutil.(*ImportWrapper).Import](http://github.com/milvus-io/milvus/internal/util/importutil.(*ImportWrapper).Import)(0xc0018b9d40, {0xc001281520?, 0x1, 0x1}, {0x2?, 0x20000000?, 0xc00053c700?, 0x40?})
        /go/src/[github.com/milvus-io/milvus/internal/util/importutil/import_wrapper.go:281](http://github.com/milvus-io/milvus/internal/util/importutil/import_wrapper.go:281) +0x935
[github.com/milvus-io/milvus/internal/datanode.(*DataNode).Import](http://github.com/milvus-io/milvus/internal/datanode.(*DataNode).Import)(0xc0007d0d80, {0x41cf1f8?, 0xc001283a40}, 0xc000df8460)
        /go/src/[github.com/milvus-io/milvus/internal/datanode/data_node.go:1098](http://github.com/milvus-io/milvus/internal/datanode/data_node.go:1098) +0x1939
[github.com/milvus-io/milvus/internal/distributed/datanode.(*Server).Import](http://github.com/milvus-io/milvus/internal/distributed/datanode.(*Server).Import)(0xf?, {0x41cf1f8?, 0xc001283a40?}, 0x10?)
        /go/src/[github.com/milvus-io/milvus/internal/distributed/datanode/service.go:379](http://github.com/milvus-io/milvus/internal/distributed/datanode/service.go:379) +0x2b
[github.com/milvus-io/milvus/internal/proto/datapb._DataNode_Import_Handler.func1](http://github.com/milvus-io/milvus/internal/proto/datapb._DataNode_Import_Handler.func1)({0x41cf1f8, 0xc001283a40}, {0x3c57240?, 0xc000df8460})
        /go/src/[github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7044](http://github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7044) +0x78
[github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor](http://github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor)({0x41cf1f8?, 0xc0012839b0?}, {0x3c57240, 0xc000df8460}, 0x41b97a0?, 0xc00000f668)
        /go/src/[github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22](http://github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22) +0x49
[github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1](http://github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1)({0x41cf1f8?, 0xc0012839b0?}, {0x3c57240?, 0xc000df8460?})
        /go/pkg/mod/[github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25](http://github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25) +0x3a
[github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1](http://github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1)({0x41cf1f8, 0xc001282600}, {0x3c57240, 0xc000df8460}, 0xc000430500?, 0xc000430520)
        /go/pkg/mod/[github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38](http://github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38) +0x16a
[github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1](http://github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1)({0x41cf1f8?, 0xc001282600?}, {0x3c57240?, 0xc000df8460?})
        /go/pkg/mod/[github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25](http://github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25) +0x3a
[github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1](http://github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1)({0x41cf1f8, 0xc001282600}, {0x3c57240, 0xc000df8460}, 0xc0012a3af0?, 0x3a42420?)
        /go/pkg/mod/[github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34](http://github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34) +0xbf
[github.com/milvus-io/milvus/internal/proto/datapb._DataNode_Import_Handler](http://github.com/milvus-io/milvus/internal/proto/datapb._DataNode_Import_Handler)({0x3ceda20?, 0xc000eea780}, {0x41cf1f8, 0xc001282600}, 0xc000fb83c0, 0xc001454cc0)
        /go/src/[github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7046](http://github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7046) +0x138
[google.golang.org/grpc.(*Server).processUnaryRPC](http://google.golang.org/grpc.(*Server).processUnaryRPC)(0xc001442380, {0x41df390, 0xc0012161a0}, 0xc0012e0120, 0xc001454e40, 0x569d3b8, 0x0)
        /go/pkg/mod/[google.golang.org/grpc@v1.46.0/server.go:1283](http://google.golang.org/grpc@v1.46.0/server.go:1283) +0xcfd
[google.golang.org/grpc.(*Server).handleStream](http://google.golang.org/grpc.(*Server).handleStream)(0xc001442380, {0x41df390, 0xc0012161a0}, 0xc0012e0120, 0x0)
        /go/pkg/mod/[google.golang.org/grpc@v1.46.0/server.go:1620](http://google.golang.org/grpc@v1.46.0/server.go:1620) +0xa1b
[google.golang.org/grpc.(*Server).serveStreams.func1.2()](http://google.golang.org/grpc.(*Server).serveStreams.func1.2())
        /go/pkg/mod/[google.golang.org/grpc@v1.46.0/server.go:922](http://google.golang.org/grpc@v1.46.0/server.go:922) +0x98
created by [google.golang.org/grpc.(*Server).serveStreams.func1](http://google.golang.org/grpc.(*Server).serveStreams.func1)
        /go/pkg/mod/[google.golang.org/grpc@v1.46.0/server.go:920](http://google.golang.org/grpc@v1.46.0/server.go:920) +0x28a
```

**Comment by yhmo** [comment]
The bug has been fixed by pr #22040 and #22041

**Comment by zhuwenxing** [comment]
Verified and passed with `2.2.0-20230208-2e4d64ec`
